view/frontend/web/js/view/checkout/summary/agreements-assigner-mixin.js

```diff
@@ -0,0 +1,57 @@
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment module (https://www.adyen.com/)
+ *
+ * Copyright (c) 2021 Adyen BV (https://www.adyen.com/)
+ * See LICENSE.txt for license details.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+define([
+    'mage/utils/wrapper',
+    'jquery'
+
+], function (wrapper, $) {
+    'use strict';
+
+    return function (agreementsAssignerAction) {
```

Comment:
```suggestion
    // Since Amazon Pay requires the checkout page to be loaded twice 
    // the agreements are confirmed automatically on the second redirect
    return function (agreementsAssignerAction) {
```

Reply:
Adding a comment for clarity here